Shared/alerts/DynamicEntityLayerAlertTarget.cpp

```diff
@@ -0,0 +1,197 @@
+  for an \l AlertCondition.
+
+  Changes to any of the dynamic entities in the layer will cause the \l AlertTarget::locationChanged
+  signal to be emitted.
+  */
+
+
+/*!
+  \brief Constructor taking an \l Esri::ArcGISRuntime::DynamicEntityLayer (\a dynamicEntityLayer).
+
+  Entities are added/updated/removed by the DynamicEntityDataSource of the DynamicEntityLayer.
+ */
+DynamicEntityLayerAlertTarget::DynamicEntityLayerAlertTarget(DynamicEntityLayer* dynamicEntityLayer) :
+  AlertTarget(dynamicEntityLayer),
+  m_dynamicEntityLayer(dynamicEntityLayer),
+  m_entityGraphics(QMap<quint64, Graphic*>{})
+{
+  // subscribe to the entity received signal from the source of the dynamic layer
+  connect(m_dynamicEntityLayer->dataSource(), &DynamicEntityDataSource::dynamicEntityReceived, this, [this](DynamicEntityInfo* info)
+  {
+    // check the new entity for a connection and mark the info as delete later
+    auto* dynamicEntity = info->dynamicEntity();
+    connectEntityGraphic(dynamicEntity);
+    info->deleteLater();
+  });
+
+  // subscribe to the observation received signal (this essentially serves as the 'update' signal for the dynamic layer type)
+  connect(m_dynamicEntityLayer->dataSource(), &DynamicEntityDataSource::dynamicEntityObservationReceived, this, [this](DynamicEntityObservationInfo* observationInfo)
+  {
+    // check the entity for a connection and mark the observation as delete later
+    auto* dynamicEntity = observationInfo->observation()->dynamicEntity();
+    connectEntityGraphic(dynamicEntity);
+    observationInfo->deleteLater();
+  });
+
+  // subscribe to the purged signal to remove any graphics from the lookup
+  connect(m_dynamicEntityLayer->dataSource(), &DynamicEntityDataSource::dynamicEntityPurged, this, [this](DynamicEntityInfo* info)
+  {
+    // ensure the graphic is actually in the lookup
+    auto* dynamicEntity = info->dynamicEntity();
+    if (m_entityGraphics.contains(dynamicEntity->entityId()))
+    {
+      // get a cleanup reference to the graphic that was stored and remove it from the lookup
+      std::unique_ptr<Graphic> graphic(m_entityGraphics[dynamicEntity->entityId()]);
+      m_entityGraphics.remove(dynamicEntity->entityId());
+
+      // signal the data has changed and rebuild the quad tree
+      rebuildQuadtree();
+      emit dataChanged();
+    }
+    info->deleteLater();
+  });
+}
+
+/*!
+  \brief Destructor.
+ */
+Dsa::DynamicEntityLayerAlertTarget::~DynamicEntityLayerAlertTarget()
+{
+}
+
+/*!
+  \brief Returns the list of \l Esri::ArcGISRuntime::Geometry which are in the \a targetArea.
+
+  \note No exact intersection tests are carried out to create this list.
+ */
+QList<Geometry> DynamicEntityLayerAlertTarget::targetGeometries(const Envelope& targetArea) const
+{
+  // if the quadtree has been built, use  it to return the set of candidate geometries
+  if (m_quadtree) { return m_quadtree->candidateIntersections(targetArea); }
```

Comment:
Please be consistent with indenting and style. There's nothing wrong with this style, but it's inconsistent with similar usages in this repo.
```suggestion
  if (m_quadtree) 
  { 
    return m_quadtree->candidateIntersections(targetArea); 
  }
```
Or omit the braces if you prefer, but not on the same line as the if statement. 

This comment applies throughout.

Reply:
Fixed throughout the file.